db/upgrade.php

```diff
@@ -131,6 +131,52 @@ function xmldb_ratingallocate_upgrade($oldversion) {
         }
 
     }
-    
+
+    if ($oldversion < 2021062900) {
+
+        // Changing type of field explanation on table ratingallocate_choices to text.
+        $table = new xmldb_table('ratingallocate_choices');
+        $field = new xmldb_field('explanation', XMLDB_TYPE_TEXT, null, null, null, null, null, 'title');
+
+        // Launch change of type for field explanation.
+        $dbman->change_field_type($table, $field);
+
+        // Ratingallocate savepoint reached.
+        upgrade_mod_savepoint(true, 2021062900, 'ratingallocate');
+    }
+
+    if ($oldversion < 2021062901) {
+
+        // Define table ratingallocate_group_choices to be created.
+        $table = new xmldb_table('ratingallocate_group_choices');
+
+        // Adding fields to table ratingallocate_group_choices.
+        $table->add_field('id', XMLDB_TYPE_INTEGER, '10', null, XMLDB_NOTNULL, XMLDB_SEQUENCE, null);
+        $table->add_field('choiceid', XMLDB_TYPE_INTEGER, '10', null, XMLDB_NOTNULL, null, null);
+        $table->add_field('groupid', XMLDB_TYPE_INTEGER, '10', null, XMLDB_NOTNULL, null, null);
+
+        // Adding keys to table ratingallocate_group_choices.
+        $table->add_key('primary', XMLDB_KEY_PRIMARY, ['id']);
+        $table->add_key('choiceid', XMLDB_KEY_FOREIGN, ['choiceid'], 'ratingallocate_choices', ['id']);
+        $table->add_key('groupid', XMLDB_KEY_FOREIGN, ['groupid'], 'groups', ['id']);
+
+        // Conditionally launch create table for ratingallocate_group_choices.
+        if (!$dbman->table_exists($table)) {
+            $dbman->create_table($table);
+        }
+
+        // Define field usegroups to be added to ratingallocate_choices.
+        $table = new xmldb_table('ratingallocate_choices');
+        $field = new xmldb_field('usegroups', XMLDB_TYPE_INTEGER, '1', null, XMLDB_NOTNULL, null, '0', 'active');
+
+        // Conditionally launch add field usegroups.
+        if (!$dbman->field_exists($table, $field)) {
+            $dbman->add_field($table, $field);
+        }
+
+        // Ratingallocate savepoint reached.
+        upgrade_mod_savepoint(true, 2021062201, 'ratingallocate');
```

Comment:
looks like the wrong save point version in here? - should be: 2021062901

Reply:
Good catch, thank you. That was a casualty of trying to do multiple DB updates in independent branches early on in the piece. Resolved in commit ea5167b.